### PR TITLE
Refactor package/application test utilities

### PR DESCRIPTION
**Pull request by a-palchikov**
## Description

Decided to give it a go while working on porting upgrade-via support from 7.x while porting package/application puller changes.
The basic idea is to have a consistent way of setting up package/application dummy data while being able to express complex configurations. This does away with text-based manifest in favor of using the Manifest values.

## Type of change

* Internal change (not necessarily a bug fix or a new feature)

## TODOs

- [x] Self-review the change
- [x] Write tests
- [x] Perform manual testing
- [ ] Address review feedback

### CODE REVIEW

**Comment by a-palchikov** [comment]
Robotest is failing with:
```
Error resolving image name 'suse-cloud/sles-15-sp2-v20201014': Could not find image or family suse-cloud/sles-15-sp2-v20201014
```
I'm guessing the image has been recycled?

**Review by bernardjkim — APPROVED** [review]
I'm not too familiar with this portion of the code, so I just took a quick look through.

File: lib/app/service/test/builder.go [hunk 1]
```diff
@@ -18,127 +18,398 @@ package test
...
+		app, err = apps.CreateApp(req.App.Manifest.Locator(), bytes.NewReader(data.Bytes()), req.App.Labels)
+		c.Assert(err, check.IsNil)
+	}
+	return app
+}
 
-	for _, p := range packageList {
-		CreateDummyPackage(loc.MustParseLocator(p), packages, c)
+// CreatePackage creates a new test package as described by the given request
+func CreatePackage(req PackageRequest, c *check.C) (pkg *pack.PackageEnvelope) {
```
[the remark this answers is not in the document]
Comment (a-palchikov) [reply lib/app/service/test/builder.go:199]: No, it returns the result of the last `PackageService.CreatePackage` call.

[the remark this answers is not in the document]
Comment (bernardjkim) [reply lib/app/service/test/builder.go:213]: Sorry, what I meant was that no callers of this function ever use the returned value.

**Comment by wadells** [comment]
> Robotest is failing with:
> 
> ```
> Error resolving image name 'suse-cloud/sles-15-sp2-v20201014': Could not find image or family suse-cloud/sles-15-sp2-v20201014
> ```


I've got a fix for this in #2606, but it looks like debian 10 is also causing issues.  I'll hopefully have a patch for that shortly.

**Pull request merged**